src/channel.js

```diff
@@ -0,0 +1,554 @@
+    const baseRows = image.pyramidBaseMetadata.Rows;
+    const baseNColumns = Math.ceil(baseTotalPixelMatrixColumns / baseColumns);
+    const baseNRows = Math.ceil(baseTotalPixelMatrixRows / baseRows);
+    for (let j = (nLevels - 1); j >= 0; j--) {
+      const columns = image.pyramidMetadata[j].Columns;
+      const rows = image.pyramidMetadata[j].Rows;
+      const totalPixelMatrixColumns = image.pyramidMetadata[j].TotalPixelMatrixColumns;
+      const totalPixelMatrixRows = image.pyramidMetadata[j].TotalPixelMatrixRows;
+      const pixelSpacing = _getPixelSpacing(image.pyramidMetadata[j]);
+      const nColumns = Math.ceil(totalPixelMatrixColumns / columns);
+      const nRows = Math.ceil(totalPixelMatrixRows / rows);
+      imageTileSizes.push([
+        columns,
+        rows,
+      ]);
+      imageGridSizes.push([
+        nColumns,
+        nRows,
+      ]);
+      imagePixelSpacings.push(pixelSpacing);
+     /*
+      * Compute the resolution at each pyramid level, since the zoom
+      * factor may not be the same between adjacent pyramid levels.
+      */
+      let zoomFactor = baseTotalPixelMatrixColumns / totalPixelMatrixColumns;
+      imageResolutions.push(zoomFactor);
+     /*
+      * TODO: One may have to adjust the offset slightly due to the
+      * difference between extent of the image at a given resolution level
+      * and the actual number of tiles (frames).
+      */
+     imageOrigins.push(offset);
+    }
+    imageResolutions.reverse();
+    imageTileSizes.reverse();
+    imageGridSizes.reverse();
+    imageOrigins.reverse();
+    imagePixelSpacings.reverse();
+
+    /** Frames may extend beyond the size of the total pixel matrix.
+     * The excess pixels are empty, i.e. have only a padding value.
+     * We set the extent to the size of the actual image without taken
+     * excess pixels into account.
+     * Note that the vertical axis is flipped in the used tile source,
+     * i.e. values on the axis lie in the range [-n, -1], where n is the
+     * number of rows in the total pixel matrix.
+     */
+
+    const imageExtents = [
+        0,                                // min X
+        -(baseTotalPixelMatrixRows + 1),  // min Y
+        baseTotalPixelMatrixColumns,      // max X
+        -1                                // max Y
+      ];
+
+    return [
+      imageExtents,  
+      imageOrigins,
+      imageResolutions,
+      imageGridSizes,
+      imageTileSizes,
+      imagePixelSpacings
+    ];
+  }
+
+  /** Adds the metadata to the metadata array of the channel
+   * @param {object} metadata
+   */
+  addMetadata(metadata){
+    this.metadata.push(metadata);
+  }
+
+  /** Gets the channel visualization/presentation parameters
+   * @returns {object} BlendingInformation
+   */
+  getBlendingInformation() {
+    return this.blendingInformation;
+  }
+
+  /** Sets the channel visualization/presentation parameters
+   * @param {Object} blendingInformation : color {number[]}, 
```

Comment:
Does this render correctly when using JSDoc?

Reply:
fixed